base/server/python/pki/server/__init__.py

```diff
@@ -1005,6 +1005,79 @@ def __repr__(self):
             return "Dogtag 9 " + self.name
         return self.name
 
+    def set_self_test(self, status, subsystem):
+
+        cmd = [
+            'pki-server'
+        ]
+
+        if status:
+            cmd.extend(['selftest-enable'])
+        else:
+            cmd.extend(['selftest-disable'])
+
+        if subsystem:
+            cmd.extend(['--subsystem', subsystem])
+
+        cmd.extend(['--instance', self.name])
+
+        subprocess.check_call(cmd)
```

Comment:
I think we should avoid going from Python to Shell then to Python again. If the code that we want can be called directly from Python that would be better. So you might want to refactor      the code out of selftest.py first.